python/semantic_kernel/agents/chat_history_handler.py

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from abc import ABC, abstractmethod
+from collections.abc import AsyncIterable
+
+from semantic_kernel.contents.chat_history import ChatHistory
+from semantic_kernel.contents.chat_message_content import ChatMessageContent
+from semantic_kernel.contents.streaming_chat_message_content import StreamingChatMessageContent
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+
+@experimental_class
+class ChatHistoryHandler(ABC, KernelBaseModel):
```

Comment:
this can just be a ABC, does not have to be a KernelBaseModel, makes it more lightweight, considering the use case I think Protocol is not usefull here